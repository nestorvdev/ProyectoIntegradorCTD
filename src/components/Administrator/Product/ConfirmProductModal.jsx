import Styles from "./Styles.module.css"

export default function ConfirmProductModal ({accion, funcionProducto, closeModalConfirm}){    
    return (
        <div className={Styles.containerCreateModal}>
            <div className={Styles.question}>?</div>
            <p>¿Está seguro que desea {accion}?</p>
            <button className={Styles.ok} onClick={funcionProducto}>Ok</button>
            <button className={Styles.cancel} onClick={closeModalConfirm}>Cancelar</button>
        </div>
    )
}