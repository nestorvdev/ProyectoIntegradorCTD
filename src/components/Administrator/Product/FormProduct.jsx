import Styles from "./Styles.module.css";
import Select from "react-select"
import Delete from "../icons/delete.svg"
import { Modal } from 'react-responsive-modal';
import 'react-responsive-modal/styles.css';
import ModalMessage from './ModalProductSucceed';

export default function FormProduct({ name, setName, selectedCategory, setSelectedCategory, address, setAddress, selectedCity, setSelectedCity, latitude, setLatitude, longitude, setLongitude, reference, setReference, qualification, setQualification, description, setDescription, selectedFeatures, setSelectedFeatures, rules, setRules, healthAndSecurity, setHealthAndSecurity, cancellationPolicy, setCancellationPolicy, imageTitle, setImageTitle, imageUrl, setImageUrl, images, setImages, categories, cities, features, setModalCreateIsOpen, enviarDatos, tituloBoton, errorCamposVacios, modalExpiredLoginIsOpen, setModalExpiredLoginIsOpen, modalExistedProductIsOpen, setModalExistedProductIsOpen }) {
    
    /*CONTROL DE COMPONENTES MEDIANTE HANDLES */
    const handleChangeName = (event) => {
        setName(event.target.value)
    }

    const handleChangeAddress = (event) => {
        setAddress(event.target.value)
    }

    const handleChangeLatitude = (event) => {
        setLatitude(event.target.value)
    }

    const handleChangeLongitude = (event) => {
        setLongitude(event.target.value)
    }

    const handleChangeReference = (event) => {
        setReference(event.target.value)
    }

    const handleChangeQualification = (event) => {
        if (event.target.value >= 1 && event.target.value <= 5) {
            setQualification({ campo: event.target.value, valido: true, error: "" })
        } else {
            setQualification({ campo: event.target.value, valido: false, error: "Ingrese una calificación del 1 al 5." })
        }
    }

    const handleChangeDescription = (event) => {
        setDescription(event.target.value)
    }

    const handleChangeRules = (event) => {
        setRules(event.target.value)
    }

    const handleChangeHealthAndSecurity = (event) => {
        setHealthAndSecurity(event.target.value)
    }

    const handleChangeCancellationPolicy = (event) => {
        setCancellationPolicy(event.target.value)
    }

    const handleChangeImageTitle = (event) => {
        setImageTitle(event.target.value)
    }

    const handleChangeImageUrl = (event) => {
        setImageUrl(event.target.value)
    }

    const handleChangeCategory = (value) => {        
        setSelectedCategory(value)
    }

    const handleChangeCity = (value) => {      
        setSelectedCity(value)
    }

    const customStyles = {
        control: () => ({
            border: "1px solid rgba(0,0,0,.4)",
            borderRadius: "7px",
            boxShadow: "0px 1px 5px rgba(0, 0, 0, 0.15)",
            display: "flex",
            justifyContent: "space-between",

        }),

        placeholder: () => ({
            padding: "9px 5px"
        }),

        valueContainer: () => ({
            display: "flex",
            minWidth: "70%",
        }),
        singleValue: () => ({
            minWidth: "70%",
            padding: "9px 5px",

        }),

        input: () => ({
            opacity: 0,
            width: 0
        }),

        option: () => ({
            color: "#31363F",
            fontWeight: 700,
            padding: "2px 15px",
            ':hover': {
                color: '#F0572D',
                cursor: 'pointer',
            },
        }),

        indicatorsContainer: () => ({
            paddingTop: "11px"
        }),

        dropdownIndicator: () => ({
            color: '#F0572D',
            padding: "0 7px",
            transition: "all 1s",
            ':hover': {
                color: '#31363F',
                cursor: 'pointer',
                opacity: "0.5"
            },
        }),
    }

    function options(arrayOptions) {
        return arrayOptions.map((valor) => {
            return {
                value: `${valor.id}`,
                label: `${valor.name}`
            };
        })
    }

    function saveSelectedFeatures(event) {
        if (event.target.checked) {
            console.log(event.target);
            setSelectedFeatures([...selectedFeatures, { id: parseInt(event.target.id), title: event.target.value}])

        } else {
            let index = selectedFeatures.findIndex(feature => feature.id === parseInt(event.target.id))            
            if (index !== -1) {
                let aux = selectedFeatures;
                aux = aux.filter((feature) => feature.id !== parseInt(event.target.id))
                setSelectedFeatures(aux)
            }
        }
    }    

    function handleIndexImageDeleted(event) {      
        deleteImage(event.target.id);    
    }

    const handleClickImage = ((event) => {
        event.preventDefault()
        if (imageTitle && imageUrl) {
            setImages([...images, { title: imageTitle, url: imageUrl }])
            setImageTitle("");
            setImageUrl("");
        }

    })

    function deleteImage(index) {
        let aux = images;
        aux = aux.filter((image) => { return image.url !== index })
        setImages(aux);
    }

    function sendData(event) {
        event.preventDefault()
    }
    
    const closeModalExpiredLogin = () => {
        setModalExpiredLoginIsOpen(false);
        sessionStorage.setItem("log", "false")
        sessionStorage.removeItem("iniciales")
        sessionStorage.removeItem("id")
        sessionStorage.removeItem("email")
        sessionStorage.removeItem("role")
        sessionStorage.removeItem("name")
        sessionStorage.removeItem("surname")
        sessionStorage.removeItem("token")
        window.location.href = "/login"
    };

    const closeModalExistedProduct = () => {
        setModalExistedProductIsOpen(false);
    };

    return (
        <form onSubmit={sendData}>
            <div className={Styles.containerBlockAdministrator}>
                <div className={Styles.blockInputs}>
                    <label htmlFor="name">Nombre de la propiedad</label>
                    <input type="text" name="name" id="name" value={name} onChange={handleChangeName} />
                </div>
                <div className={Styles.blockInputs}>
                    <label htmlFor="category">Categoría</label>
                    <Select
                        onChange={(newValue) => handleChangeCategory(newValue)}
                        options={options(categories)}
                        placeholder="Seleccionar categoria"
                        styles={customStyles}
                        getOptionValue={option => option.value}
                        value={selectedCategory}
                    />
                </div>
            </div>
            <div className={Styles.containerBlockAdministrator}>
                <div className={Styles.blockInputs}>
                    <label htmlFor="address">Dirección</label>
                    <input type="text" name="address" id="address" value={address} onChange={handleChangeAddress} />
                </div>
                <div className={Styles.blockInputs}>
                    <label htmlFor="city">Ciudad</label>
                    <Select
                        onChange={(newValue) => handleChangeCity(newValue)}
                        options={options(cities)}
                        placeholder="Seleccionar ciudad"
                        styles={customStyles}
                        getOptionValue={(option) => option.value}
                        value={selectedCity}
                    />
                </div>
            </div>
            <div className={Styles.containerBlockAdministrator}>
                <div className={Styles.blockInputs}>
                    <label htmlFor="latitude">Latitud</label>
                    <input type="number" name="latitude" id="latitude" value={latitude} onChange={handleChangeLatitude} />
                </div>
                <div className={Styles.blockInputs}>
                    <label htmlFor="longitude">Longitud</label>
                    <input type="number" name="longitude" id="longitude" value={longitude} onChange={handleChangeLongitude} />
                </div>
            </div>
            <div className={Styles.containerBlockAdministrator}>
                <div className={Styles.blockInputs}>
                    <label htmlFor="reference">Referencia</label>
                    <input type="text" name="reference" id="reference" value={reference} onChange={handleChangeReference} />
                </div>
                <div className={Styles.blockInputs}>
                    <label htmlFor="qualification">Calificación</label>
                    <input type="number" name="qualification" id="qualification" value={qualification.campo !== "" ? parseInt(qualification.campo) : ""} onChange={handleChangeQualification} />
                    <div className={Styles.error}>{qualification.error}</div>
                </div>
            </div>
            <div className={Styles.description}>
                <label htmlFor="description">Descripción</label>
                <textarea name="description" id="description" maxLength="500" placeholder="Escribir aquí" value={description} onChange={handleChangeDescription} />
            </div>

            <div className={Styles.containerCheckbox}>
                <h3>Agregar atributos</h3>
                {features.map((option, index) => {
                    return (
                        <label onClick={saveSelectedFeatures}>
                            <input
                                onChange={event => saveSelectedFeatures(event)}
                                type="checkbox"
                                checked={selectedFeatures.find(feature => feature.id === option.id)}
                                id={index + 1}
                                value={option.name}
                            />
                            {option.name}
                        </label>
                    )
                })}
            </div>
            <div className={Styles.containerPoliticsPrincipal}>
                <h3>Políticas del producto</h3>
                <h6>(Separar las políticas por coma)</h6>
                <div className={Styles.containerPolitics}>
                    <div className={Styles.politics}>
                        <h5>Normas de la casa</h5>
                        <label>Descripción</label>
                        <textarea name="rules" id="rules" placeholder="Escribir aquí" value={rules} onChange={handleChangeRules} />
                    </div>
                    <div className={Styles.politics}>
                        <h5>Salud y seguridad</h5>
                        <label>Descripción</label>
                        <textarea name="healthAndSecurity" id="healthAndSecurity" placeholder="Escribir aquí" value={healthAndSecurity} onChange={handleChangeHealthAndSecurity} />
                    </div>
                    <div className={Styles.politics}>
                        <h5>Políticas de cancelación</h5>
                        <label>Descripción</label>
                        <textarea name="cancellationPolicy" id="cancellationPolicy" placeholder="Escribir aquí" value={cancellationPolicy} onChange={handleChangeCancellationPolicy} />
                    </div>
                </div>
            </div>
            <div className={Styles.containerImages}>
                <h3>Cargar imágenes</h3>
                <h6>(Las imágenes se cargan haciendo click en +)</h6>
                <div className={Styles.containerBlockAdministratorImage}>
                    <div className={Styles.blockInputsImages}>
                        <div className={Styles.image}>
                            <label htmlFor="titleImage">Título</label>
                            <input type="text" name="titleImage" id="titleImage" value={imageTitle} onChange={handleChangeImageTitle} />
                        </div>
                        <div className={Styles.image}>
                            <label htmlFor="urlImage">URL</label>
                            <input type="text" name="urlImage" id="urlImage" value={imageUrl} onChange={handleChangeImageUrl} />
                        </div>
                    </div>
                    <button onClick={handleClickImage}>+</button>
                </div>
                <div className={Styles.insertedImages}>
                    {images.map((image, index) => {
                        return (
                            <div className={Styles.insertedImageAndButton}>
                                <div className={`${Styles.insertedImage}`}>
                                    <input type="text" name="titleImage" id={index + 1} className={Styles.imageItem} value={image.title} />
                                    <input type="text" name="urlImage" id={index + 1} className={Styles.imageItem} value={image.url} />
                                </div>
                                <div className={Styles.deleteImage} ><img src={Delete} alt="icon delete" id={image.url} onClick={handleIndexImageDeleted} /></div>
                            </div>
                        )
                    })}
                </div>
            </div>
            <div>
                <button onClick={(e) => enviarDatos(e)} id={Styles.buttonCreateProduct} type="submit">{tituloBoton}</button>
            </div>
            <div className={Styles.error}>{errorCamposVacios}</div>
            <Modal open={modalExpiredLoginIsOpen} onClose={closeModalExpiredLogin} center>
                <ModalMessage title={"El login expiró"} message={"Por favor inicie sesión nuevamente."} closeModal={closeModalExpiredLogin} icon="X" />
            </Modal>
            <Modal open={modalExistedProductIsOpen} onClose={closeModalExistedProduct} center>
                <ModalMessage title={"El producto ya existe."} message={"Por favor, verifique los datos e intente nuevamente."} closeModal={closeModalExistedProduct} icon="X" />
            </Modal>
        </form >
    )
}