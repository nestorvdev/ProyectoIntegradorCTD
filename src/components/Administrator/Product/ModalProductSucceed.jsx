
import Styles from "./Styles.module.css"

export default function ProductModalSucceed ({title, message, closeModal, icon}){   
    return (
        <div className={Styles.containerCreateModal}>
            <div className={icon === "X" ? Styles.cruz : null}> {icon === "X"? icon : <img src={icon} alt="icon"/>}</div>
            <h3 >{title}</h3>
            <p>{message}</p>
            <button onClick={closeModal}>Ok</button>
        </div>
    )
}