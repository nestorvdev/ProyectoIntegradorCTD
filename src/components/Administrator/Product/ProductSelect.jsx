import React, { useState, useEffect } from "react";
import Styles from "./Styles.module.css";
import Select from "react-select";
import StylesApp from "../../../App.module.css"
import axios from "axios";

function ProductSelect({ handleProduct }) {
    const [errorMessage, setErrorMessage] = useState("");
    const [products, setProducts] = useState([]);   
    const [chosenProduct, setChosenProduct] = useState({
        id: null,
        name: "",
        description: "",
        latitude: null,
        longitude: null,
        qualification: null,
        favorite: null,
        reference: "",
        category: { id: 1, title: "", description: "", url: "" },
        city: { id: 1, name: "", country: "" },
        images: [{ id: null, title: "", url: "", productId: null }],
        features: [{ id: 1, title: "", state: null }],
        rules: "",
        health: "",
        politics: "",
        address: "",
    });
    console.log(chosenProduct);
    const baseURL = "http://worldguestbooking.com.ar:8080/";

    useEffect(() => {
        axios
            .get(baseURL + "products/all")
            .then((response) => {
                setProducts(response.data);               
            })          
            .catch((error) => {
                setErrorMessage(error.message);
            });
        
    }, []);
    console.log(chosenProduct);

    const options = products.map((p) => {
        return {
            key: p.id,
            label: `${p.id} ${p.name}`,
            value: p,
        };
    })

    const customStyles = {
        control: () => ({
            border: "1px solid rgba(0,0,0,.4)",
            borderRadius: "7px",
            boxShadow: "1px 2px 4px rgba(0, 0, 0, 0.25)",
            display: "flex",
            justifyContent: "space-between",
            backgroundColor: "white",
            width: "43vw",
            minWidth: "250px",
            maxWidth: "590px",
            color: "black",
            '@media all and (max-width:415px)': {
                width: "100%"
            }
        }),

        placeholder: () => ({
            padding: "10px 10px"
        }),

        valueContainer: () => ({
            display: "flex",
            minWidth: "70%",
        }),
        singleValue: () => ({
            minWidth: "70%",
            padding: "10px 10px",

        }),

        input: () => ({
            opacity: 0,
            width: 0
        }),

        option: () => ({
            color: "#31363F",
            fontWeight: 700,
            padding: "2px 15px",
            ':hover': {
                color: '#F0572D',
                cursor: 'pointer',
            },
        }),

        indicatorsContainer: () => ({
            paddingTop: "11px"
        }),

        dropdownIndicator: () => ({
            color: '#F0572D',
            padding: "0 7px",
            transition: "all 1s",
            ':hover': {
                color: '#31363F',
                cursor: 'pointer',
                opacity: "0.5"
            },

        }),
    }
    
    let handleChange = (option) => {        
        setChosenProduct(option.value);  
      
    }

    return (
        <section className={`${StylesApp.delimiter} ${Styles.containerPrincipal}`}>
            <div className={`${StylesApp.delimiterChild} ${Styles.containerForm} ${Styles.containerProductSelect}`}>
                <p className={Styles.titleProductSelect}>Seleccione el nombre del producto para modificar</p>
                <div className={Styles.selectProductBox}>
                    <Select                     
                        options={options}
                        placeholder="Seleccionar producto"
                        styles={customStyles}
                        isSearchable
                        onChange={(newValue) => handleChange(newValue)}                        
                    />
                    <button onClick={() => { handleProduct(chosenProduct) }} className={`${Styles.buttonSearchProduct} ${Styles.button}`}>Buscar</button>
                </div>
            </div>
        </section>
    )
}

export default ProductSelect;