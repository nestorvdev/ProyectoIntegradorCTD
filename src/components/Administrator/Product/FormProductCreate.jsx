import React, { useState } from "react";
import StylesApp from "../../../App.module.css"
import Styles from "./Styles.module.css";
import { Modal } from 'react-responsive-modal';
import 'react-responsive-modal/styles.css';
import ModalProductSucceed from './ModalProductSucceed';
import ConfirmProductModal from "./ConfirmProductModal";
import FormProduct from "./FormProduct";
import { AxiosCrearProducto } from "../../../axiosCollection/Product/AxiosProduct";
import tildeOk from "../icons/tildeOk.svg"

export default function FormProductCreate({ categories, cities, features, titleModal, messageModal }) {
    const [name, setName] = useState("");
    const [selectedCategory, setSelectedCategory] = useState()
    const [address, setAddress] = useState("");
    const [selectedCity, setSelectedCity] = useState()
    const [latitude, setLatitude] = useState();
    const [longitude, setLongitude] = useState();
    const [reference, setReference] = useState("");
    const [qualification, setQualification] = useState({campo:"", valido:false, error:""});
    const [description, setDescription] = useState("");
    const [selectedFeatures, setSelectedFeatures] = useState([]);
    const [rules, setRules] = useState("");
    const [healthAndSecurity, setHealthAndSecurity] = useState("");
    const [cancellationPolicy, setCancellationPolicy] = useState("");
    const [imageTitle, setImageTitle] = useState()
    const [imageUrl, setImageUrl] = useState()
    const [images, setImages] = useState([])
    const [errorProduct, setErrorProduct] = useState("");
    const [errorCamposVacios, setErrorCamposVacios] = useState("")
    const [modalProductSucceedIsOpen, setModalProductSucceedIsOpen] = useState(false)
    const [modalConfirmIsOpen, setModalConfirmIsOpen] = useState(false)
    const [modalExpiredLoginIsOpen, setModalExpiredLoginIsOpen] = useState(false)
    const [modalExistedProductIsOpen, setModalExistedProductIsOpen] = useState(false) 
    
    const openModalSucceed = (() => {
        setModalProductSucceedIsOpen(true)
    })
    const closeModalSucceed = () => {
        setModalProductSucceedIsOpen(false);
        window.location.href = "/"
    };

    const openModalConfirm = (e) => {
        e.preventDefault();
        if (name && description && latitude && longitude && address && qualification.valido && reference && selectedCategory && selectedCity && rules && healthAndSecurity && cancellationPolicy && images.length > 0 && selectedFeatures.length > 0) {
            setModalConfirmIsOpen(true)
            setErrorCamposVacios("")
        } else {
            setErrorCamposVacios("Por favor complete todos los campos")
        }

    }

    const closeModalConfirm = () =>{
        setModalConfirmIsOpen(false)
    }

    const openModalExpiredLogin = (() => {
        setModalExpiredLoginIsOpen(true)
    })

    const openModalExistedProduct = (() => {
        setModalExistedProductIsOpen(true)
    })

    const crearProducto = () =>{
        closeModalConfirm()
        AxiosCrearProducto(name, description, latitude, longitude, address, qualification.campo, reference, selectedCategory.value, selectedCity.value, rules, healthAndSecurity, cancellationPolicy, images, selectedFeatures, setErrorProduct, openModalSucceed, openModalExpiredLogin, openModalExistedProduct)
        console.log(errorProduct);
    }

    return (
        <section className={`${StylesApp.delimiter} ${Styles.containerPrincipal}`}>
            <div className={`${StylesApp.delimiterChild} ${Styles.containerForm}`}>
                <FormProduct
                    name={name} setName={setName}
                    selectedCategory={selectedCategory} setSelectedCategory={setSelectedCategory}
                    address={address} setAddress={setAddress}
                    selectedCity={selectedCity} setSelectedCity={setSelectedCity}
                    latitude={latitude} setLatitude={setLatitude}
                    longitude={longitude} setLongitude={setLongitude}
                    reference={reference} setReference={setReference}
                    qualification={qualification} setQualification={setQualification}
                    description={description} setDescription={setDescription}
                    selectedFeatures={selectedFeatures} setSelectedFeatures={setSelectedFeatures}
                    rules={rules} setRules={setRules}
                    healthAndSecurity={healthAndSecurity} setHealthAndSecurity={setHealthAndSecurity}
                    cancellationPolicy={cancellationPolicy} setCancellationPolicy={setCancellationPolicy}
                    imageTitle={imageTitle} setImageTitle={setImageTitle}
                    imageUrl={imageUrl} setImageUrl={setImageUrl}
                    images={images} setImages={setImages}
                    categories={categories} cities={cities} features={features}
                    setModalProductSucceedIsOpen={setModalProductSucceedIsOpen}
                    enviarDatos={openModalConfirm} tituloBoton={"Crear"}
                    errorCamposVacios={errorCamposVacios}
                    modalExpiredLoginIsOpen={modalExpiredLoginIsOpen} setModalExpiredLoginIsOpen={setModalExpiredLoginIsOpen}
                    modalExistedProductIsOpen={modalExistedProductIsOpen} setModalExistedProductIsOpen={setModalExistedProductIsOpen}
                />
                <Modal open={modalConfirmIsOpen} onClose={closeModalConfirm} center>
                    <ConfirmProductModal accion="crear el producto" setModalConfirmIsOpen={setModalConfirmIsOpen} funcionProducto={crearProducto} closeModalConfirm={closeModalConfirm}/>
                </Modal>
                <Modal open={modalProductSucceedIsOpen} onClose={closeModalSucceed} center>
                    <ModalProductSucceed title={titleModal} message={messageModal} closeModal={closeModalSucceed} icon={tildeOk}/>
                </Modal>
            </div>
        </section>
    )
}