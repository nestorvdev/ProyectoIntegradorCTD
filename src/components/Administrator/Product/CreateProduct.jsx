import React, { useState, useEffect } from "react";
import StylesApp from "../../../App.module.css";
import Spinner from "../../spinner/Spinner";
import FormProductCreate from "./FormProductCreate";
import TitleBar from "../../Product/TitleBar";
import { AxiosGetCategories, AxiosGetCities, AxiosGetFeatures } from "../../../axiosCollection/Product/AxiosProduct.jsx"

function CreateProduct(props) {
    const [optionsCategories, setOptionsCategories] = useState([])
    const [optionsCities, setOptionsCities] = useState([])
    const [optionsFeatures, setOptionsFeatures] = useState([])
    const [loading, setLoading] = useState(true);
    const [errorMessage, setErrorMessage] = useState("");
    const title = "¡Gracias!";
    const message = "Se ha creado exitosamente el producto";

    useEffect(() => {
        AxiosGetCategories(setLoading, setOptionsCategories, setErrorMessage)
        AxiosGetCities(setLoading, setOptionsCities, setErrorMessage)
        AxiosGetFeatures(setLoading, setOptionsFeatures, setErrorMessage)
    }, [])    

    return (
        (errorMessage && loading) ?
            <section className={StylesApp.delimiter}>
                <h1>{errorMessage}</h1>
            </section>
            :
            <section>
                {loading ? (
                    <Spinner />
                ) : (
                    <>
                        <TitleBar name="Administración de productos" goBack={props.goBack} />
                        <section className={StylesApp.delimiter}>
                            <div className={StylesApp.delimiterChild}>
                                <h1>Crear producto</h1>
                                <FormProductCreate categories={optionsCategories} cities={optionsCities} features={optionsFeatures} titleModal={title} messageModal={message} />
                            </div>
                        </section>
                    </>
                )}
            </section>
    )
}

export default CreateProduct;