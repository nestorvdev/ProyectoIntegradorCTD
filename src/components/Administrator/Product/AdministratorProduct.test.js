import React from "react";
import { shallow, mount, render } from "enzyme";
import "@testing-library/jest-dom";
import CreateProduct from "./CreateProduct";
import ConfirmProductModal from "./ConfirmProductModal";
import FormProduct from "./FormProduct";
import FormProductCreate from "./FormProductCreate";
import FormProductUpdate from "./FormProductUpdate";
import ModalProductSucceed from "./ModalProductSucceed";
import ProductSelect from "./ProductSelect";
import UpdateSelect from "./UpdateProduct";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';


Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

//PROBANDO ARCHIVO CREATE-PRODUCT

describe("Probando el componente <CreateProduct/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<CreateProduct/>)
    });

    it("Deberia mostrar <CreateProduct/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

//PROBANDO ARCHIVO CONFIRM-PRODUCT-MODAL

describe("Probando el componente <ConfirmProductModal/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<ConfirmProductModal/>)
    });

    it("Deberia mostrar <ConfirmProductModal/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

//PROBANDO ARCHIVO FORM-PRODUCT

describe("Probando el componente <FormProduct/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<FormProduct/>)
    });

    it("Deberia mostrar <FormProduct/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia llamar a la funcion handleClickImage al hacer click", () => {
        wrapper.find('.containerCheckbox label').simulate("click")
        expect(wrapper).toMatchSnapshot();
    });


})

//PROBANDO ARCHIVO FORM-PRODUCT-CREATE

describe("Probando el componente <FormProductCreate/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<FormProductCreate/>)
    });

    it("Deberia mostrar <FormProductCreate/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})  

//PROBANDO ARCHIVO FORM-PRODUCT-UPDATE

describe("Probando el componente <FormProductUpdate/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<FormProductUpdate/>)
    });

    it("Deberia mostrar <FormProductUpdate/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})  

//PROBANDO ARCHIVO MODAL-PRODUCT-SUCCEED

describe("Probando el componente <ModalProductSucceed/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<ModalProductSucceed/>)
    });

    it("Deberia mostrar <ModalProductSucceed/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

//PROBANDO ARCHIVO PRODUCT-SELECT

describe("Probando el componente <ProductSelect/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<ProductSelect/>)
    });

    it("Deberia mostrar <ProductSelect/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})  

//PROBANDO ARCHIVO UPDATE-SELECT

describe("Probando el componente <UpdateSelect/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<UpdateSelect/>)
    });

    it("Deberia mostrar <UpdateSelect/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})  
