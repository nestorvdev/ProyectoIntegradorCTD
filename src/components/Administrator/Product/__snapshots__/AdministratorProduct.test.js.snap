// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <ConfirmProductModal/> Deberia mostrar <ConfirmProductModal/> correctamente 1`] = `
<div
  className="containerCreateModal"
>
  <div
    className="question"
  >
    ?
  </div>
  <p>
    ¿Está seguro que desea 
    ?
  </p>
  <button
    className="ok"
  >
    Ok
  </button>
  <button
    className="cancel"
  >
    Cancelar
  </button>
</div>
`;

exports[`Probando el componente <CreateProduct/> Deberia mostrar <CreateProduct/> correctamente 1`] = `
<section>
  <Spinner />
</section>
`;

exports[`Probando el componente <FormProductCreate/> Deberia mostrar <FormProductCreate/> correctamente 1`] = `
<section
  className="delimiter containerPrincipal"
>
  <div
    className="delimiterChild containerForm"
  >
    <FormProduct
      address=""
      cancellationPolicy=""
      description=""
      enviarDatos={[Function]}
      errorCamposVacios=""
      healthAndSecurity=""
      images={Array []}
      modalExistedProductIsOpen={false}
      modalExpiredLoginIsOpen={false}
      name=""
      qualification={
        Object {
          "campo": "",
          "error": "",
          "valido": false,
        }
      }
      reference=""
      rules=""
      selectedFeatures={Array []}
      setAddress={[Function]}
      setCancellationPolicy={[Function]}
      setDescription={[Function]}
      setHealthAndSecurity={[Function]}
      setImageTitle={[Function]}
      setImageUrl={[Function]}
      setImages={[Function]}
      setLatitude={[Function]}
      setLongitude={[Function]}
      setModalExistedProductIsOpen={[Function]}
      setModalExpiredLoginIsOpen={[Function]}
      setModalProductSucceedIsOpen={[Function]}
      setName={[Function]}
      setQualification={[Function]}
      setReference={[Function]}
      setRules={[Function]}
      setSelectedCategory={[Function]}
      setSelectedCity={[Function]}
      setSelectedFeatures={[Function]}
      tituloBoton="Crear"
    />
    <ForwardRef
      center={true}
      onClose={[Function]}
      open={false}
    >
      <ConfirmProductModal
        accion="crear el producto"
        closeModalConfirm={[Function]}
        funcionProducto={[Function]}
        setModalConfirmIsOpen={[Function]}
      />
    </ForwardRef>
    <ForwardRef
      center={true}
      onClose={[Function]}
      open={false}
    >
      <ProductModalSucceed
        closeModal={[Function]}
        icon="tildeOk.svg"
      />
    </ForwardRef>
  </div>
</section>
`;

exports[`Probando el componente <FormProductUpdate/> Deberia mostrar <FormProductUpdate/> correctamente 1`] = `
<section
  className="delimiter containerPrincipal"
>
  <div
    className="delimiterChild containerForm"
  >
    <FormProduct
      address=""
      cancellationPolicy=""
      description=""
      enviarDatos={[Function]}
      errorCamposVacios=""
      healthAndSecurity=""
      images={Array []}
      modalExpiredLoginIsOpen={false}
      name=""
      qualification={
        Object {
          "campo": "",
          "error": "",
          "valido": false,
        }
      }
      rules=""
      selectedCategory={
        Object {
          "id": "",
          "name": "",
        }
      }
      selectedCity={
        Object {
          "id": "",
          "name": "",
        }
      }
      selectedFeatures={Array []}
      setAddress={[Function]}
      setCancellationPolicy={[Function]}
      setDescription={[Function]}
      setHealthAndSecurity={[Function]}
      setImageTitle={[Function]}
      setImageUrl={[Function]}
      setImages={[Function]}
      setLatitude={[Function]}
      setLongitude={[Function]}
      setModalExpiredLoginIsOpen={[Function]}
      setModalProductSucceedIsOpen={[Function]}
      setName={[Function]}
      setQualification={[Function]}
      setReference={[Function]}
      setRules={[Function]}
      setSelectedCategory={[Function]}
      setSelectedCity={[Function]}
      setSelectedFeatures={[Function]}
      tituloBoton="Modificar"
    />
    <ForwardRef
      center={true}
      onClose={[Function]}
      open={false}
    >
      <ConfirmProductModal
        accion="modificar el producto"
        closeModalConfirm={[Function]}
        funcionProducto={[Function]}
        setModalConfirmIsOpen={[Function]}
      />
    </ForwardRef>
    <ForwardRef
      center={true}
      onClose={[Function]}
      open={false}
    >
      <ProductModalSucceed
        closeModal={[Function]}
        icon="tildeOk.svg"
      />
    </ForwardRef>
  </div>
</section>
`;

exports[`Probando el componente <ModalProductSucceed/> Deberia mostrar <ModalProductSucceed/> correctamente 1`] = `
<div
  className="containerCreateModal"
>
  <div
    className={null}
  >
     
    <img
      alt="icon"
    />
  </div>
  <h3 />
  <p />
  <button>
    Ok
  </button>
</div>
`;

exports[`Probando el componente <ProductSelect/> Deberia mostrar <ProductSelect/> correctamente 1`] = `
<section
  className="delimiter containerPrincipal"
>
  <div
    className="delimiterChild containerForm containerProductSelect"
  >
    <p
      className="titleProductSelect"
    >
      Seleccione el nombre del producto para modificar
    </p>
    <div
      className="selectProductBox"
    >
      <ForwardRef
        isSearchable={true}
        onChange={[Function]}
        options={Array []}
        placeholder="Seleccionar producto"
        styles={
          Object {
            "control": [Function],
            "dropdownIndicator": [Function],
            "indicatorsContainer": [Function],
            "input": [Function],
            "option": [Function],
            "placeholder": [Function],
            "singleValue": [Function],
            "valueContainer": [Function],
          }
        }
      />
      <button
        className="buttonSearchProduct button"
        onClick={[Function]}
      >
        Buscar
      </button>
    </div>
  </div>
</section>
`;

exports[`Probando el componente <UpdateSelect/> Deberia mostrar <UpdateSelect/> correctamente 1`] = `
<section>
  <Spinner />
</section>
`;
